src/azure-cli-testsdk/azure/cli/testsdk/utilities.py

```diff
@@ -210,6 +211,38 @@ def process_response(self, response):
         return response
 
 
+class EmailAddressReplacer(RecordingProcessor):
+    """Replace email address like xxx@microsoft.com with test@example.com"""
+
+    EMAIL_REPLACEMENT = 'test@example.com'
+
+    def _replace_email_address(self, text):
+        pattern = r'[\w.%#+-]+[%40|@|_]microsoft.com'
```

Comment:
Any source for this regex?

Reply:
The source is me😂 Any issue with this pattern?

Reply:
I am curious how `%#+-` are chosen. Could you provide some examples? I understand there are email addresses like `Isaac.Newton@microsoft.com` which contains `[\w.]`.

Reply:
for example, `v-jingszhang@microsoft.com` contains `-`. But for `%#+`, I don't have any sample on top of my mind